plugin/message_plane/rpc_server.py

```
@@ -0,0 +1,683 @@
+                if rr.search(str(got or "")):
+                    matched_regex.append(ev)
+            return matched_regex
+
+        return None
+
+    def _apply_binary_op(self, left: List[Dict[str, Any]], right: List[Dict[str, Any]], *, op: str) -> Optional[List[Dict[str, Any]]]:
+        if op not in ("merge", "intersection", "difference"):
+            return None
+        left_keys = [self._dedupe_key(x) for x in left]
+        right_keys = [self._dedupe_key(x) for x in right]
+        set_left = set(left_keys)
```

Ruff (0.14.13)

Local variable `set_left` is assigned to but never used

Remove assignment to unused variable `set_left`

(F841)